caikit/runtime/servicers/runtime_info_servicer.py

```diff
@@ -0,0 +1,68 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have pylint ignore Class XXXX has no YYYY member so that we can use gRPC enums.
+# pylint: disable=E1101
+
+# Standard
+from typing import Dict
+from importlib.metadata import version as pip_version
+
+# Third Party
+import grpc
+
+# First Party
+import alog
+
+# Local
+from caikit.core import MODEL_MANAGER
+from caikit.interfaces.runtime.data_model import (
+    RuntimeInfoStatusResponse,
+    RuntimeInfoStatusResponseDict,
+)
+from caikit.config import get_config
+
+log = alog.use_channel("RI-SERVICR-I")
+
+
+class RuntimeInfoServicerImpl:
+    """This class contains the implementation of all of the RPCs that are required to run a
+    service in Model Mesh as a Model-Runtime."""
+
+    def GetRuntimeInfo(self, request, context):
+        """Get information on versions of libraries and server from config"""
+        version_dict = {}
+        # TODO: how to get library version for extensions?
+        # version_dict["caikit_nlp_version"] = pip_version("caikit_nlp")
+        version_dict["caikit_version"] = pip_version("caikit")
+        # TODO: why is cls.config != get_config() within this method --> fails with error  AttributeError: type object 'RuntimeHTTPServer' has no attribute 'config'
+        print("DOES VERSIONING EXIST", get_config().runtime.versioning)
+
+        if get_config().runtime.versioning:
+            print("GET_CONFIG", get_config().runtime.versioning)
+            # TODO: how to get library that is being run -- aka get caikit_nlp part dynamically
+            version_dict.update(get_config().runtime.versioning)
```

Comment:
@anhuong have you looked into importlib.metadata.packages_distributions for the heavy lifting here? or figuring out what's installed?
https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata

Something like this might get us 90% there:
```
from importlib.metadata import version, packages_distributions

for p in packages_distributions().keys():
    if "caikit" in p:
        version_dict[p] = version(p)
    # or anything extra the user requested:
    if p in request.packages:
        version_dict[p] = version(p)
```

Reply:
Oh yeah we were just looking at that yesterday as well but that seems like a package that is new in 3.10 so we won't be able to use since we support 3.8, 3.9

Reply:
Ah right. I think the `importlib_metadata` package exists for this reason, it looks like it's a full backport of `importlib.metadata`

Reply:
Interesting! How does that work to `import importlib_metadata` for <3.10 and do `import importlib.metadata` for >=3.10?

Reply:
@anhuong you could probably import `importlib_metadata` in all cases, or..
- Do a try/catch
```
try:
    from importlib.metadata import version
except ImportError:
    from importlib_metadata import version
```
- grab the python version off sys
```
if sys.version_info < (3, 11):
    from importlib_metadata import version
else:
    from importlib.metadata import version
```

Reply:
Doing `importlib_metadata` didn't work in py311 for me but the alternative is great!

Reply:
> you could probably import importlib_metadata in all cases, or..

i see what you're saying now, took a little while but it finally clicked 😅